### PR TITLE
fix: Azure managed disk tier pricing

### LINKED ISSUE #3870: bug(azure): Premium SSD PVs underpriced 16x — Disk Mount meter overwrites Disk meter in convertMeterToPricings
**Describe the bug**
`managed-csi-premium` PVs are underpriced by ~16× in v1.120.3. PR #3778 correctly fixed SKU detection (`skuname` lookup), but the `premium_ssd` price itself is wrong — `convertMeterToPricings()` matches both `P4 LRS Disk` ($5.28/mo) and `P4 LRS Disk Mount` ($0.32/mo) via the `"P4 "` prefix filter. The mount meter overwrites the disk meter, so only the mount fee is used for pricing.

Standard SSD is unaffected because Azure does not publish a separate `E4 LRS Disk Mount` meter.

**To Reproduce**
1. Run OpenCost on AKS with PVs using the `managed-csi-premium` StorageClass (`disk.csi.azure.com`, `skuname: Premium_LRS`)
2. Query `pv_hourly_cost` for any of those PVs

**Expected behavior**
`pv_hourly_cost` reflects the `P4 LRS Disk` rate. For Central US: $5.2795/mo → $2.26e-04/GiB/hr (~$0.165/GiB/month).

**Screenshots**
Here you can see the price change when we implemented `v1.120.3` for the skuname fix.
Image
Image

Here are the rates as shown on Azure Pricing.
Image

**Which version of OpenCost are you using?**
1.120.3

**Additional context**

- Cloud: Azure (AKS, Central US)
- Observed rate: `1.40e-05/GiB/hr` ($0.010/GiB/month) — matches `P4 LRS Disk Mount` ($0.32/mo ÷ 730 ÷ 32)
- Correct rate: `2.26e-04/GiB/hr` — from `P4 LRS Disk` ($5.2795/mo ÷ 730 ÷ 32)

### PR DESCRIPTION
**Pull request by sawyerward**
## Description
Refactors Azure managed disk pricing to be tier-aware and fixes incorrect pricing behavior in rate-card ingestion. Current pricing assumes a linear per-GB across disk sizes, but that is not accurate for Microsoft pricing.

## Related Issues
Fixes #3870

## User Impact
Azure AKS users with managed disks (especially Premium SSD classes like `Premium_LRS` ) will get more accurate PV/disk pricing based on actual disk-tier meters, with correct fallback behavior when specific tier meters are missing.

Windows VM pricing keys are preserved during pricing-map cleanup.

This is not a breaking change; it is a correctness improvement for Azure cost attribution/pricing behavior.

## Testing
- Added regression tests for:
  - disk mount overwrite prevention,
  - managed-disk key filtering preserving Windows keys,
  - tier-aware PV pricing and fallback logic,
  - tier-aware disk cost lookup.
- Added focused unit tests in `disktiers_test.go` for meter parsing, SKU resolution, tier selection, conversion helpers, and managed-disk key detection.
- Ran:

```bash
go test ./pkg/cloud/azure

### CODE REVIEW

File: pkg/cloud/azure/disktiers.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+		{Name: "S6", SizeGiB: 64},
+		{Name: "S10", SizeGiB: 128},
+		{Name: "S15", SizeGiB: 256},
+		{Name: "S20", SizeGiB: 512},
+		{Name: "S30", SizeGiB: 1024},
+		{Name: "S40", SizeGiB: 2048},
+		{Name: "S50", SizeGiB: 4096},
+		{Name: "S60", SizeGiB: 8192},
+		{Name: "S70", SizeGiB: 16384},
+		{Name: "S80", SizeGiB: 32767},
```
Comment (ameijer) [inline pkg/cloud/azure/disktiers.go:61]: @sawyerward are you concerned about these going out of date/becoming stale?

**Review by ameijer — COMMENTED** [review]
@sawyerward have you tested this on your clusters? we do not have an OC azure test cluster to play around with

File: pkg/cloud/azure/disktiers.go (hunk 1 above)
Comment (sawyerward) [reply pkg/cloud/azure/disktiers.go:61]: It is possible, but right now the biggest one is 32 TiB so they'd have to release a drive bigger than that. It's unfortunate that they don't have a linear per-GB price so we have to maintain some sort of mapping.

**Comment by sawyerward** [comment]
> @sawyerward have you tested this on your clusters? we do not have an OC azure test cluster to play around with

We have it running in some of our sandbox clusters. The numbers are looking good but I'll likely let it run over the weekend to get better numbers.

**Comment by sawyerward** [comment]
I haven't had time to wrap up all my testing but here's the results for StandardSSD.

Here we are testing the `managed` and `managed-csi` storage classes which both map to `StandardSSD_ZRS`. We tested in `eastus2` region. This shows how when a PV isn't using the full disk size (e.g. 65 GiB/128 GiB), the total cost when multiplied by the PV size equals the total disk tier size (128 GiB = $14.40) and is the same total monthly cost as if you're using the full 128 GiB PV. 

Azure bills based on the provisioned tier size not on the PV size itself. The effective rate is computed per-requested-GiB so that allocation multiplied by tier rate equals the full tier price.

https://azure.microsoft.com/en-us/pricing/details/managed-disks/

Screenshot 2026-07-10 at 11 36 27 AM
Screenshot 2026-07-10 at 11 37 28 AM

**Review by ameijer — APPROVED** [review]
going to give this a shot @sawyerward , can you pull out of draft so we can merge?

**Comment by sawyerward** [comment]
> going to give this a shot @sawyerward , can you pull out of draft so we can merge?

I could do that. I'm currently trying to test the ZRS edge case. It's not showing up in the rate card API so I'd have to set up an SP with permissions for the price sheet API, etc. Wouldn't get to that until next week. If you're okay without those specific test results, let's get it merged. Right now the ZRS lookups log like:

```2026-07-10T18:48:29.977567696Z WRN Persistent Volume tier pricing not found for eastus2,premium_ssd,ZRS,P6 size 64 (ZRS); falling back to class rate```
